pkg/git/secret.go

```diff
@@ -180,3 +185,31 @@ func ParseUsernameAndPassword(secret string) (string, string) {
 	}
 	return "", ""
 }
+
+func NewGitSecretProvider(client client.Client, namespace string, gitSource *v1alpha1.GitSource) (*SecretProvider, error) {
+	if gitSource.Spec.SecretRef == nil {
+		return NewSecretProvider(nil), nil
+	}
+	coreSecret := &corev1.Secret{}
+	namespacedSecretName := types.NamespacedName{Namespace: namespace, Name: gitSource.Spec.SecretRef.Name}
+	err := client.Get(context.TODO(), namespacedSecretName, coreSecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch the secret object")
```

Comment:
Maybe, we should check not found and return `missing secret`, this way user will understand he is referring some secret in `GitSource`, but doesn't have secrets already created

Reply:
> Looks like we are setting WATCH_NAMESPACE in operator.yaml.

thanks for pointing this out. looks like this change got lost as part of my "rebase dance" - fixed: 4245095 

>  this way user will understand he is referring some secret in GitSource, but doesn't have secrets already created

it's a question how this will be managed from the UI side. I think that UI should be responsible for setting correct secret name to the GitSource CR. This is something I need to clarify with UI guys

Reply:
Yes UI would be responsible for creating the secret and setting it in the `secretRef`. If it's not there, assume the user didn't put it intentionally at that point.

Reply:
I do agree with you UI is responsible for that, but it might be possible that there is some spelling mistake while configuring secret from UI, then in that case for user it's quite difficult to understand what's going wrong even after seeing failed status